pkg/dohdec/test/dnsUtils.ava.js

```diff
@@ -51,6 +51,27 @@ test('makePacket - subnet & ecs = 16', t => {
   t.is(options.sourcePrefixLength, ecs)
 })
 
+test('makePacket - dnssec disabled', t => {
+  const pkt = DNSutils.makePacket({name: 'foo'})
+  const dns = packet.decode(pkt)
+  t.false(dns.flag_ad)
+  t.false(dns.flag_cd)
+})
+
+test('makePacket - dnssec', t => {
+  const pkt = DNSutils.makePacket({name: 'foo', dnssec: true})
+  const dns = packet.decode(pkt)
+  t.true(dns.flag_ad)
+  t.false(dns.flag_cd)
+})
+
+test('makePacket - dnssec with cd=1', t => {
+  const pkt = DNSutils.makePacket({name: 'foo', dnssec: true, dnssecCd: true})
+  const dns = packet.decode(pkt)
+  t.true(dns.flag_ad)
+  t.true(dns.flag_cd)
+})
+
```

Comment:
does `{dnssec: false, dnssecCd: true}` make sense?  If not, should it be an error?

Reply:
Good question. I can't find the answer in the DNSSEC RFCs, but that combination works OK if we set `cd=1` with that combination and run the following test:

```js
test('DoH - dnssec disabled with cd=1', async t => {
  const doh = new DNSoverHTTPS()
  const r = await doh.lookup('dnssec-failed.org', {
    json: false,
    dnssec: false,
    dnssecCd: true,
  })
  t.truthy(r)

  t.is(r.rcode, 'NOERROR')
  t.false(r.answers.some(a => a.type === 'RRSIG'))
  t.true(r.flag_cd)
})
```

So I'm going to change the code to set `cd=1` solely based on the value of `dnssecCheckingDisabled` (fka `dnssecCd`), regardless of the value of `dnssec`.

Reply:
you haven't made this change yet, right?

Reply:
I did: https://github.com/hildjj/dohdec/pull/38/commits/acd49694a83825a461bdff55e4a4a63ca7a4bbef

Reply:
Sorry, i must have been looking at an outdated thing.  I see it now!